app/models/bottleneck_event.rb

```diff
@@ -31,11 +32,14 @@ def self.generate_future_events(obj)
       event.timestamp     = result.delete(:timestamp)
       event.context_data  = e[:definition].merge(result)
       event.message       = event.substitute(event.message)
-      event.save
-      adds += 1
+      if event.save
+        adds += 1
+      else
+        _log.warn("#{log_message_uniq_prefix} failed with '#{event.errors.full_messages.join(', ')}'")
```

Comment:
This looks good. In this case the warning message is appropriate because we're not raising an error.